artmanflow/steps/common.py

```diff
@@ -0,0 +1,367 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+import errno
+import tempfile
+import uuid
+import os
+import subprocess
+import sys
+from ruamel import yaml
+
+
+class ConfigUtils:
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def read_config(yaml_file_path=None, yaml_str=None, execution_id=None):
+        config = ConfigUtils._read_config(yaml_file_path, yaml_str)
+        if ('execution_id' not in config) and execution_id:
+            config['execution_id'] = execution_id
+        return config
+
+    @staticmethod
+    def _read_config(yaml_file_path=None, yaml_str=None):
+        if yaml_str:
+            return yaml.safe_load(yaml_str)
+        elif yaml_file_path:
+            with open(yaml_file_path) as config_file:
+                return yaml.safe_load(config_file.read())
+        elif len(sys.argv) <= 1:
+            return yaml.safe_load(sys.stdin)
+        else:
+            with open(sys.argv[1]) as config_file:
+                return yaml.safe_load(config_file.read())
+
+    @staticmethod
+    def dump_config(config, output):
+        if isinstance(output, str):
+            with open(output, 'w') as config_file:
+                yaml.round_trip_dump(config, config_file)
+        else:
+            yaml.round_trip_dump(config, output)
+            output.flush()
+
+    @staticmethod
+    def generate_id(prefix):
+        return "%s%s" % (prefix, str(uuid.uuid4())[24:])
+
+    @staticmethod
+    def artifact_name():
```

Comment:
We do not generally write methods like this in Python. (I would be happy to patch this, but I need more insight into what the class actually does.)

Reply:
This is a getter method without "get" prefix. I took this naming pattern from another artman file: `artman/utils/protoc_utils.py`.